clients/src/main/java/org/apache/kafka/common/requests/ElectPreferredLeadersRequest.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.requests;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.types.ArrayOf;
+import org.apache.kafka.common.protocol.types.Field;
+import org.apache.kafka.common.protocol.types.Schema;
+import org.apache.kafka.common.protocol.types.Struct;
+import org.apache.kafka.common.utils.CollectionUtils;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.kafka.common.protocol.CommonFields.TOPIC_NAME;
+import static org.apache.kafka.common.protocol.types.Type.INT32;
+
+public class ElectPreferredLeadersRequest extends AbstractRequest {
+
+    private static final String TIMEOUT_KEY_NAME = "timeout";
+    private static final String TOPIC_PARTITIONS_KEY_NAME = "topic_partitions";
+    private static final String PARTITIONS_KEY_NAME = "partitions";
+
+    public static final Schema ELECT_PREFERRED_LEADERS_REQUEST_V0 = new Schema(
+            new Field(TOPIC_PARTITIONS_KEY_NAME, ArrayOf.nullable(
```

Comment:
Now that we have the .json file, the idea is to leverage the generated class when constructing the request objects to avoid hand-written code. See ElectPreferredLeadersResponse.json for how we converted CreateTopics to use the generated code.

Reply:
I don't see CreateTopics converted in #5893, only in #5972 which hasn't been merged yet. I can do the work for making the new request and response used the generated messages, but there are further changes needed, to support it. For example `AbstractResponse.parseResponse()` need an extra parameter for the version. These changes are in #5972, but make me wonder if I'm going astray here, and actually we need #5972 to be merged before I could really use it. Please advise.

Reply:
Hi @tombentley,

Thanks for being patient here.  I agree that there are a few changes in #5972 that you might want.  However, they're actually extremely minimal.  I think you could easily pull them into this PR if you wanted to.

Basically the changes are:
* Edit `import-control.xml` to allow some Java packages to use classes from `org.apache.kafka.common.message`
* `AbstractResponse#parseResponse` should take a version argument.  The version isn't stored in the response class, but it is used to parse the response struct correctly.

If you want, I can pull just these two changes into a separate PR.  Or you can just pull them into this PR.

By the way, I have rebased #5972 on trunk and it is now ready for review.  Take a look.